packages/lab/src/stepped-tracker/TrackerStep/TrackerStep.tsx

```diff
@@ -19,46 +21,27 @@ import trackerStepCss from "./TrackerStep.css";
 
 const withBaseName = makePrefixer("saltTrackerStep");
 
-type State = "default" | "completed";
-
-type StateWithActive = State | "active";
+type IconOptions = "default" | "completed" | "warning" | "error";
 
 export interface TrackerStepProps extends ComponentPropsWithoutRef<"li"> {
   /**
+   * The slug of the icon to be displayed in the TrackerStep
+   */
+  icon?: IconOptions | null;
```

Comment:
We can't call this icon, we already use icon in Toast for a JSX component

Reply:
`iconSlug` feels more appropriate, but maybe more obtuse?
Assuming we can't use `adornment` for similar reasons? (`startAdornment` and `endAdornment` are used in a few places for JSX components)

Reply:
Have set all the values to `TBC_PROP_NAME` (easy to find/replace once we decide tomorrow)